lib/puppet/util/windows/user.rb

```diff
@@ -22,6 +22,14 @@ def admin?
   # https://msdn.microsoft.com/en-us/library/windows/desktop/ee207397(v=vs.85).aspx
   SECURITY_MAX_SID_SIZE = 68
 
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/ms681385(v=vs.85).aspx
+  # These error codes indicate successful authentication but failure to
+  # logon for a separate reason
+  ERROR_ACCOUNT_RESTRICTION = 1327
+  ERROR_INVALID_LOGON_HOURS = 1328
+  ERROR_INVALID_WORKSTATION = 1329
+  ERROR_ACCOUNT_DISABLED    = 1331
```

Comment:
I was hoping to find a definitive MSDN page with all the possible return values for `LogonUser` rather than just the exhaustive error code list.  But it looks like that page doesn't exist.  Hmmph.